### DISCUSSION TITLE
Wayland support: Known bugs

### DISCUSSION THREAD

**Discussion by nbolton** [deskflow/deskflow#7499] [General]
# Wayland support

 Wayland has been supported since v1.17.0

Older systems may need to have libportal built, if you want to tinker with the code and you're missing libportal: Follow the [Static `libportal` build instructions](https://github.com/deskflow/deskflow/discussions/7499#discussioncomment-11027553)

# Desktop Environment Support
  - V1 Input Capture / Remote Desktop  (unable to use clipboard)
    - GNOME (v46+)
    - KDE Plasma (6.1+) 
  - V2 Input Capture / Remote Desktop (clipboard access provided)
    - Gnome (v50+)
    - KDE Plasma(6.7.1+)

# Portals backends that do not support Input-Capture  (Server)
You need to have a portal with Input-Capture support (the one in use depends on your DE) for server mode to work under wayland
   - gtk (https://github.com/flatpak/xdg-desktop-portal-gtk/issues/494)
   - wlroots (https://github.com/emersion/xdg-desktop-portal-wlr/issues/278)
   - lxde (https://github.com/lxqt/xdg-desktop-portal-lxqt/issues/42)
   - elementary (https://github.com/elementary/portals/issues/117)
   - deepin (https://github.com/orgs/linuxdeepin/discussions/10659)
   - xapp (https://github.com/linuxmint/xdg-desktop-portal-xapp/issues/21)
   - hyperland(https://github.com/hyprwm/xdg-desktop-portal-hyprland/issues/259)
   - cosmic (https://github.com/pop-os/xdg-desktop-portal-cosmic/issues/197)
   - niri (https://github.com/YaLTeR/niri/issues/823)
 
# Portals backends that do not support Remote Desktop (Client)
You need to have a portal with RemoteDesktop support (the one in use depends on your DE) for client mode to work under wayland
 - gtk (No issue found)
 - wlroots(https://github.com/emersion/xdg-desktop-portal-wlr/issues/2)
 - lxde (No Issue found)
 - elementry(https://github.com/elementary/portals/issues/149)
 - deepin (https://github.com/linuxdeepin/xdg-desktop-portal-dde,  No issue found)
 - xapp (no issue found)
 - hyperland(https://github.com/hyprwm/xdg-desktop-portal-hyprland/issues/252)
 - cosmic (https://github.com/pop-os/xdg-desktop-portal-cosmic/issues/23)
 - niri (https://github.com/YaLTeR/niri/issues/390)

# Wayland support: Known bugs

These will eventually be raised as actual issues:
- ~~Bug: System tray not working on Wayland (endless retry loop)~~
- ~~Bug: Prevent computer from sleeping no longer works on Wayland~~ (#8947) 
- ~~Bug: Unable to leave server screen from right hand side (Ubuntu only?)~~

# Issues we need to fix 

- Hotkey support 
  -  libportal PR: https://github.com/flatpak/libportal/pull/226
  -  Our Impl https://github.com/deskflow/deskflow/pull/9858
- #7632

## Issues that have to be fixed by upstream 

These bugs need to be fixed in a project Wayland use depends upon:
 - Mouse is not hidden on Wayland server when leaving screen (Report to your compositor)
 - Mouse is not hidden on Wayland client when leaving screen (libEi bug ? ) 
 - Modifiers not sent to clients when on host screen https://github.com/deskflow/deskflow/issues/8437
 - libportal does not support hotkey portal (https://github.com/flatpak/libportal/pull/226)

Previous discussion: https://github.com/deskflow/deskflow/discussions/7456
Related: https://github.com/input-leap/input-leap/discussions/1976

**Comment by enzious** [comment]
> * Bug: Copy and paste doesn't seem to work on Wayland

If you're going the libportal route...

It looks like the infrastructure for it is there for Gnome. My initial interpretation is that it's only available for RemoteDesktop? It's odd that they tied it to just RemoteDesktop.

However, it looks like this may be blocked on KDE for a while:
https://github.com/KDE/xdg-desktop-portal-kde/blame/master/src/remotedesktop.cpp#L216

I just went down the whole rabbit hole: deskflow -> libportal -> xdg-desktop-portal -> xdg-desktop-portal-kde

**Reply by gabeklavans** [reply]
Does this newish data control portal help with the clipboard issue at all?

https://gitlab.freedesktop.org/wayland/wayland-protocols/-/blob/main/staging/ext-data-control/ext-data-control-v1.xml

https://lists.freedesktop.org/archives/wayland-devel/2024-December/043920.html

**Reply by nbolton** [reply]
Surfaced on the issue: https://github.com/deskflow/deskflow/issues/8031#issuecomment-2581307209

**Comment by sithlord48** [comment]
>It looks like the infrastructure for it is there for Gnome. My initial interpretation is that it's only available for RemoteDesktop? It's odd that they tied it to just RemoteDesktop.

InputCapture is not in the kde 6.0 release but RemoteDesktop is. Not Adding it to Input-Capture was an oversight for sure.

**Comment by nbolton** [comment]
## Building for Ubuntu 24.04 (static `libportal` workaround)

> [!WARNING]
> This is a hack to workaround for Ubuntu 24.04 only having `libportal` 0.7
> We hope that Ubuntu 24.10 will soon follow Debian Trixie and update to `libportal` 0.8

### 1. Checkout at v1.17.0 tag:
```
git clone git@github.com:deskflow/deskflow.git
git checkout tags/v1.17.0
```

### 2. Apply patch to drop min `libei` to 1.2
```diff
diff --git a/cmake/Libraries.cmake b/cmake/Libraries.cmake
index 6f930875..0bd7a5a3 100644
--- a/cmake/Libraries.cmake
+++ b/cmake/Libraries.cmake
@@ -13,7 +13,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-set(LIBEI_MIN_VERSION 1.3)
+set(LIBEI_MIN_VERSION 1.2)
 set(LIBPORTAL_MIN_VERSION 0.8)
 
 macro(configure_libs)
```

### 3. Configure with static `libportal` 0.8
> [!TIP]
> If you already checked out and built, run: `rm -r build` to blow away your build dir.
```
./scripts/install_deps.py --meson-no-system libportal --meson-static libportal --subprojects
cmake -B build -G Ninja -DSYSTEM_LIBPORTAL=OFF
```

### 4. Verify in cmake output that you're using the local `libportal` build
> [!TIP]
> Make sure it says `libportal.a` (static lib) and not `libportal.so` (dynamic lib).
> On Ubuntu 24.10 this will say libei 1.3.0
```
-- libei version: 1.2.1
-- Using local subproject libportal
-- libportal library file: /home/nick/deskflow/build/meson/subprojects/libportal/libportal/libportal.a
```

### 6. Build and static link against local `libportal` build
```
cmake --build build
```

### 7. Verify that we did not link to a dynamic `libportal`
```
ldd build/bin/deskflow-server | grep libportal
```

### 8. Run the GUI and verify that it works
```
./build/bin/deskflow
```

**Comment by homelab-00** [comment]
> Copy and paste doesn't work on Wayland

This can be fixed temporarily using [ClipCascade](https://github.com/Sathvik-Rao/ClipCascade) as a clipboard sync.

_Sorry if this isn't the right place to post this._

**Reply by sithlord48** [reply]
> Sorry if this isn't the right place to post this.

Could be a whole discussion on its own but since you have already said it here ill just respond with 

> Note: Linux systems use different display servers, primarily X11 and Wayland. Wayland can also include XWayland, which supports X sessions. The program automatically detects the display server and selects the appropriate interface, either GUI or CLI, based on the environment.


You can also use [KDE Connect](https://kdeconnect.kde.org) to sync clipboards and wayland is supported if your using a plasma session.

**Reply by homelab-00** [reply]
KDE Connect was my first thought as well, but it just didn't work between my Fedora 41 KDE and Windows 10 machines.

**Reply by JayXon** [reply]
How does ClipCascade support Wayland? Can deskflow do the same thing?

**Reply by gabeklavans** [reply]
> How does ClipCascade support Wayland? Can deskflow do the same thing?

It uses wl-copy and wl-paste, which are just command line apps for interacting with clipboard. I think Deskflow wants to do everything though portals, however, for which the support for clipboard interaction isn't fully there yet to my knowledge.

**Reply by nixternal** [reply]
> KDE Connect was my first thought as well, but it just didn't work between my Fedora 41 KDE and Windows 10 machines.

Did you retry KDE Connect? I just set it up on a whim and disabled all plugins except for clipboard sharing & it works.

**Comment by dirad** [comment]
> Options for enabling Wayland support:
> 
>     1. [Download Deskflow v1.17](https://github.com/deskflow/deskflow/releases/tag/v1.17.0)
> 
>     2. [Download Synergy v1.16](https://symless.com/synergy/download/other) (if you purchased Synergy)

are you sure about these ?

No matter what configuration I try, the logs always point to the same error 
```
INFO: starting core server process (desktop mode)

[2025-02-11T14:56:08] INFO: using ei screen for wayland
[2025-02-11T14:56:08] ERROR: failed to initialize input capture session, quitting: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.freedesktop.portal.InputCapture” on object at path /org/freedesktop/portal/desktop
```

I've tried Deskflow [1.19.0] , Synergy [1.16-1, 1.17, 3.2.1], and barrier. 
They all said the same.

Is the problem i'm using hyprland ?        
If so - is there a way around it?

**Reply by sithlord48** [reply]
Yes wait for hyperland to have support in its portal backend.. the bug is in the description (https://github.com/hyprwm/xdg-desktop-portal-hyprland/issues/259)

**Reply by zhurai** [reply]
looks like this is still active: https://github.com/hyprwm/xdg-desktop-portal-hyprland/pull/268
- NOTE: not merged yet/draft
- someone wrote these instructions https://gist.github.com/stauersbol/235884919606ea3584c0db48d9c94129 on how to try it out prior to merge

**Comment by xobs** [comment]
I've been using this on Fedora 42 (since it kept installing itself and uninstalling Synergy 3) with Plasma, and it works mostly alright. I'm looking forward to getting copy-paste working again. Hopefully soon!

I am seeing the issue where I get "blinking" if I ram my cursor against the edge of the screen. I have my Mac on the left, but if I move the cursor to the top, bottom, or right of the screen then it flickers as if it's... trying to warp the cursor somewhere? As I move it towards the edge of the screen it blinks or jumps back and forth by one pixel towards the edge of the screen. However, I believe this is the Plasma issue mentioned above.

There are a number of issues I've found that I haven't seen reported anywhere else:

* Keys or mouse buttons occasionally get stuck. This is particularly problematic on the Mac, where it seems like the Command (Alt) key keeps getting stuck. The solution is to open Deskflow and click the "Apply" button, which seems to restart the server.
* It seems as though mouseover events are given to windows that are in the background. Sometimes. For example, if Firefox is not the foreground window, then I can see events given to e.g. the Tree Style Tab sidebar. But interestingly enough, only when the cursor is over the decorative shadow of the border of the overlapping window, and only to the chrome, not to the page itself. That is, tabs get highlighted, but only when the cursor is near the edge of the window on top.
* Similarly, shaking the cursor in a circle while it is on another display causes it to become visible on my main Wayland machine, resulting in a visible cursor both on the Wayland machine and the Mac controlling it.
* It's possible to control another machine even when the main machine is locked. Is this considered an issue?

**Reply by gabeklavans** [reply]
I've experienced all of these issues as well (but the first one about getting keys stuck happens only very rarely for me).

Same setup, F42 server and macos client on the side

**Reply by sithlord48** [reply]
The blinking when on the screen edge should be fixed in the current continuous version

**Reply by gabeklavans** [reply]
> The blinking when on the screen edge should be fixed in the current continuous version 

Does the fix you're mentioning fully address https://github.com/deskflow/deskflow/issues/7632?

Just wondering, it would be nice to track there

**Reply by sithlord48** [reply]
That is correct. I've added it to the description.

**Comment by WardBrian** [comment]
I've been having some trouble with media keys on my mouse/kb. My goal is for these to always trigger on the host (Windows) PC, so even if my mouse happens to be on my second screen they will still correctly play/pause the music on my host.

In previous (X-based) set ups, I accomplished something like this using a custom hotkey. However now, if I set up a hotkey based on these keys, the connection refuses to re-activate between the two computers until I disable or remove the custom hotkey. 

The log during this period is full of
```
core       [2025-08-20T14:20:21] - WARNING - failed to connect to server: Timed out
core       [2025-08-20T14:20:24] - WARNING - failed to connect to server: Timed out
core       [2025-08-20T14:20:27] - WARNING - failed to connect to server: Timed out
core       [2025-08-20T14:20:30] - WARNING - failed to connect to server: Timed out
```

**Reply by WardBrian** [reply]
Oh, I found this thread for an old email and missed it being Deskflow-specific. My issue is in Synergy 3 -- is there a better place to report such a thing?

**Reply by nbolton** [reply]
> My issue is in Synergy 3 -- is there a better place to report such a thing?

Please let tech support know about Synergy issues: https://symless.com/synergy/contact

They make sure it's raised on the Synergy internal issue tracker.

If you could also try Deskflow to see if it happens there, that would be helpful, since Synergy 3 is downstream of Deskflow.

**Comment by hmoffatt** [comment]
I'm not sure if this is Wayland-specific, but I notice that if am working on a remote screen for a few minutes, the server's own screen may lock/sleep according to the normal desktop power settings, even though I'm actively using the server keyboard and mouse.

This causes the server to stop working as a server, so the next keyboard/mouse event wakes it up again (and prompts for capture permission) and then I resume working. But it's quite disruptive.

I see this in the logs when it happens
```
[2025-08-22T14:59:02] INFO: ei: Disconnected by EIS
[2025-08-22T14:59:02] WARNING: disconnected from eis, will afterwards commence attempt to reconnect
```

I'm using Deskflow 1.22.0 on Linux as the server, and on Windows as the client. This wasn't happening when I was using Barrier on X11 as the server.

**Reply by nbolton** [reply]
Permission dialogs are a new feature in Wayland (Portals).

We have a very janky and insecure workaround for that:
https://github.com/deskflow/accept-portal-dialog

**Reply by hmoffatt** [reply]
Thanks, I understand the permission dialog and I've been following your workaround. My question is why the server screensaver is kicking in when the server is in active use?

**Reply by nbolton** [reply]
> My question is why the server screensaver is kicking in when the server is in active use?

Oh, I see. Missed that. For me it happens when the cursor is off the screen. Perhaps we need to send a DBus keep wake/suppress sleep message when the cursor is off the screen. 

Maybe tangentially related:
- #8827

Temporary hack that I use on GNOME:
```
gnome-session-inhibit --inhibit idle --reason "Prevent lock" sleep 999999
```

KDE equivalent:
```
systemd-inhibit --what=idle --why="Prevent screen lock" sleep infinity
```

**Reply by sithlord48** [reply]
i plan to do this over dbus and portals once i can call them from QDbus correctly . there is a portal for this

**Comment by linaori** [comment]
On Wayland Fedora KDE since a recent release (or recent deskflow update maybe?) I no longer have to confirm the permissions every time the app starts. I think this relates to "Permissions dialog shows every time"?

Deskflow: 1.24.0.0 (56a1bd68)
Qt: 6.9.2
System: KDE Flatpak runtime
Session: KDE (wayland)

Operating System: Fedora Linux 42
KDE Plasma Version: 6.4.4
KDE Frameworks Version: 6.17.0
Qt Version: 6.9.1
Kernel Version: 6.16.7-200.fc42.x86_64 (64-bit)
Graphics Platform: Wayland
Processors: 8 × Intel® Core™ Ultra 7 268V
Memory: 32 GiB of RAM (30.9 GiB usable)
Graphics Processor: Intel® Graphics
Manufacturer: LENOVO
Product Name: 21NSCTO1WW
System Version: ThinkPad X1 Carbon Gen 13

**Reply by sithlord48** [reply]
v 1.24.0 is the first relase to support to support Remote desktop token persistance allowing us to store the token the client uses. So we no longer need to ask every time. 

This is not yet supported for Input Capture (server) We are waiting for that portal to support a restore token. (https://github.com/flatpak/xdg-desktop-portal/pull/1768)

**Comment by berglh** [comment]
Is `Bug: Mouse is not hidden on Wayland server or client when leaving screen` meant to be fixed as it's crossed out or is this wont fix? This is still visible on server when controlling the client and vice-versa (which is fine I guess).

Running continuous release on:

Deskflow: 1.25.0.102 (fdb5405e)
Qt: 6.9.2
System: Ubuntu 25.10
Session: ubuntu:GNOME (wayland)

**Reply by sithlord48** [reply]
Yes but its only fixed for servers (at least with KDE). The client fix needs to come from upstream. I've updated above to reflect that. This needs is something that has to happen in the compositor / libEI  communication somehow. You may need to report a bug to GNOME's compositor.

**Reply by berglh** [reply]
I see, this is not fixed on the server on Wayland with Gnome/Mutter so thanks for updating that :)

**Comment by souravdas142** [comment]
Any news or status for swaywm (nowhere for swaywm any issue is tracking) ? 

window manager: swaywm
OS: Arch Linux
Deskflow: 1.25.0
Display server: wayland[2026-01-06T20:30:19.021] INFO: starting client process (desktop mode)
[2026-01-06T20:30:19.023] INFO: running command: /usr/bin/deskflow-core client
[2026-01-06T20:30:19.149] INFO: using ei screen for wayland
[2026-01-06T20:30:19.295] NOTE: started client
[2026-01-06T20:30:19.297] IPC: connecting to '192.168.29.89': 192.168.29.89:24800
[2026-01-06T20:30:19.304] ERROR: failed to initialize remote desktop session: GDBus.Error:org.freedesktop.DBus.Error.InvalidArgs: No such interface “org.freedesktop.portal.RemoteDesktop”

getting the below error

```bash
[2026-01-06T19:50:52.565] INFO: starting server process (desktop mode)
[2026-01-06T19:50:52.569] INFO: core config file: /home/sourav/.config/Deskflow/deskflow-server.conf
[2026-01-06T19:50:52.569] INFO: running command: /usr/bin/deskflow-core server
[2026-01-06T19:50:52.699] IPC: started server, waiting for clients
[2026-01-06T19:50:52.704] ERROR: failed to initialize input capture session, quitting: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.freedesktop.portal.InputCapture” on object at path /org/freedesktop/portal/desktop
```

**EDIT:** for completeness

When running in client mode below error also getting (as i planned for windows server and archbox client)

```bash
[2026-01-06T20:30:19.021] INFO: starting client process (desktop mode)
[2026-01-06T20:30:19.023] INFO: running command: /usr/bin/deskflow-core client
[2026-01-06T20:30:19.149] INFO: using ei screen for wayland
[2026-01-06T20:30:19.295] NOTE: started client
[2026-01-06T20:30:19.297] IPC: connecting to '192.168.29.89': 192.168.29.89:24800
[2026-01-06T20:30:19.304] ERROR: failed to initialize remote desktop session: GDBus.Error:org.freedesktop.DBus.Error.InvalidArgs: No such interface “org.freedesktop.portal.RemoteDesktop”
```

**Reply by sithlord48** [reply]
nope go file a bug on their tracker and provide a link and ill add it to the list above.

**Reply by souravdas142** [reply]
Sure. @sithlord48

**Reply by jestarray** [reply]
I have a similar issue with sway. Since sway uses wlroots, I presume when its fixed in wlroots, it will also be fixed in sway?

**Comment by GOKORURI007** [comment]
I've written [pynergy](https://github.com/GOKORURI007/pynergy) as a temporary workaround for using Wayland compositors as clients, while we wait for proper RemoteDesktop portal support in the compositors themselves.

**Reply by sithlord48** [reply]
That is better suited for a new discussion in the ["Show and Tell"](https://github.com/deskflow/deskflow/discussions/categories/show-and-tell) section.

**Reply by GOKORURI007** [reply]
Thanks for the suggestion! I've moved the introduction of pynergy to a new post in the Show and Tell section here: #9490 .

**Comment by petersilva** [comment]
on ubuntu 26.04 using default everything... (Wayland + Gnome 50, deskflow 1.25)  All the versions are supposed to be high enough, but it still does not work... am I missing a dependency or something?

```
host% deskflow-core server
initial settings file: /home/user/.config/Deskflow/Deskflow.conf
[2026-02-20T10:34:56.319] INFO: using ei screen for wayland
[2026-02-20T10:34:56.331] ERROR: failed to initialize input capture session, quitting: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.freedesktop.portal.InputCapture” on object at path /org/freedesktop/portal/desktop
[2026-02-20T10:34:56.331] IPC: started server, waiting for clients
[2026-02-20T10:34:56.331] WARNING: current primary screen is not entered on shutdown
[2026-02-20T10:34:56.332] WARNING: failed to enable system idle sleep
[2026-02-20T10:34:56.332] NOTE: stopped server
host% 
```

**Reply by sithlord48** [reply]
yes you are
>  ERROR: failed to initialize input capture session, quitting: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.freedesktop.portal.InputCapture” on object at path /org/freedesktop/portal/desktop

but also might need to run the gui to click ok to allow the program to use it

**Comment by nmofonseca** [comment]
Hello,

Let me start to five a big shout out and thank you for this amazing project!!!

I was wondering if anyone can help me, since I am not sure what I am doing wrong.

I am using CahcyOS (Arch Linux-based distro) as the server, and I enabled the wayland support by enabling. I believe ChachyOS has KDE plasma which uses Wayland.

<img width="645" height="367" alt="image" src="https://github.com/user-attachments/assets/ecea4e09-c5ee-451f-a12d-e2b3dba97ccc" />

The client is running Windows 11. 

I have been trying CachyOS more recently, and before it was Windows 11 to Windows 11, and the clipboard worked just fine.

The deskflow version, both in Linux and Windows, is 1.26.0

I can't see any errors in the log files. Here is an e.g. when I try to copy something from the server to the client:

Is there anything else required?


EDIT: 

YES, there was something missing, what a muppet! For some reason, I just assumed wl-clipboard was installed. After checking with Pacman, I confirmed that wl-clipboard indeed wasn't installed.

I just installed wl-clipboard, restarted and voilá, copy-paste is freaking working!!

I am just leaving this here for others not commit the same stupid mistake.

**Comment by nmofonseca** [comment]
Hello

I hope someone can help.

I did read over the comments and the issues logged here. According to the log, the flickering/blinking issues are resolved, but I have been facing an issue since yesterday, after installing wl-clipboard for clipboard support.

I can use any menus; they flicker, and sometimes I see some flashing on the screen.

Not sure if this recording shows it very well, but there it goes.

[Screencast_20260226_104419.webm](https://github.com/user-attachments/assets/bb37b650-7eba-4b18-b778-c8c65570bbdf)

I am absolutely not knowledgeable enough to troubleshoot this. But I will give it a go if I can get some help.

For now, I will just disable the wl-clipboard support and enable it when I need it.

The client is Windows 11, and the Server is CachyOS

**Reply by sithlord48** [reply]
This is focus stealing bug mentionded in the dialog where you activate it.. there is not a release of wl-clipboard w/o this so you will need to built it yourself or use the wl-clipboard-git package.

**Reply by nmofonseca** [reply]
@sithlord48 ,

First of all, thanks for taking the time to reply to me.

I wouldn't be able to probably build it myself,  I can see wl-clipboard-git is availabe in the AUR repositories, so will give it a go with that :)

I will let you know how it goes.

**Reply by sithlord48** [reply]
it works ok even w/ the newest version is not perfect and will be removed as soon as the next release of libportal comes out w/ clipboard support. it(wl-clipboard) can also cause the host to hold to the port causeing the core to not restart correctly in some cases if that happen try doing `kill -9 $(pidof wl-copy)`

**Reply by nmofonseca** [reply]
@sithlord48 ,

Worked likea charm, flickering, blinking gone.

Can I just ask one more thing? Are there any downsides to using wl-clipboard-git instead of wl-clipboard?

Not sure if this happens to a lot of people. Should this be added to the logged issues here and register wl-clipboard-git as a fix/workaround? 

Thank you so much for all yor help

**Reply by sithlord48** [reply]
nope but it doesn't work right on all desktops its supported in that it might work for you but it maynot.

**Comment by nbolton** [comment]
**Hyprland news!**

Hyprland support is almost ready to land for Input Capture portal (will let Deskflow/Synergy work on Hyprland server):
- https://github.com/hyprwm/xdg-desktop-portal-hyprland/pull/268

Edit: Remote Desktop portal (for Hyprland client) has two PRs but not sure which will land:
- https://github.com/hyprwm/xdg-desktop-portal-hyprland/pull/308
- https://github.com/hyprwm/xdg-desktop-portal-hyprland/pull/402

**Comment by kameqdev** [comment]
I also encountered a bug specific to Wayland.

**Server:** Windows 11
**Client:** EndeavourOS (Wayland)

While using the Right Alt key on the Windows PC keyboard to type special characters on the Linux client (such as ą, ć, ó), these characters are always typed capitalized (Ą, Ć, Ó). 
Switching the client session to X11 completely eliminates this issue, so it's strictly related to how Wayland handles the input.

**Comment by eleucarlos** [comment]
I want to share a working solution for the AltGr/accent bug on Wayland — we solved it on Hyprland and it may help here.

## Context

On Hyprland (client mode), Synergy 3.6.3 connects via the `RemoteDesktop` portal + libei. Two things were broken:

1. **No RemoteDesktop portal backend** on Hyprland — fixed with a small user-level bridge (`hypr-kdeconnect-portal`).
2. **AltGr accents were broken** — exactly the issue reported above (`ą ć ó` coming out capitalized / plain). 

## The accent fix (Mod5)

The Windows server sends AltGr as **keycode 84** (`ISO_Level3_Shift`), which maps to **Mod5** (xkb index 7) in the US-Intl keymap. The bridge wasn't reflecting keycode 84 in the `modifiers` mask, so the compositor never activated Mod5.

Fix in the bridge (`wayland_input.cpp`):

```cpp
if (isDown(84))            // ISO_Level3_Shift (AltGr)
    mask |= 1u << 7;       // Mod5
```

Plus:
- Use **US-Intl** keymap (`us` + variant `intl`) when creating the keymap.
- Translate left Alt (56) and 84 → right Alt (100) so AltGr works like on Windows.
- **AltGr release timing:** Windows sends AltGr *up* before the letter *up* (`100D 24D 100U 24U`), which makes the compositor swallow the accent. Hold the AltGr release until the letter is released.

Result: `á é í ó ú`, `ç`, `ą ć ó` all type correctly on Hyprland client.

## Also relevant: shortcuts on Wayland

Hyprland doesn't fire keybinds with **SUPER** from virtual keyboards (issues [#3125](https://github.com/hyprwm/Hyprland/issues/3125)/[#8537](https://github.com/hyprwm/Hyprland/issues/8537)) — only ALT/CTRL work. Workaround: add ALT variants of the SUPER shortcuts in the Hyprland config.

Full working guide: https://github.com/omacom/omarchy/discussions/10097

**Reply by sithlord48** [reply]
This is very specific to hyperland it might work better in something like their wiki ?
